provider/file/file.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/bank-vaults/secret-init/model"
+)
+
+const ProviderName = "file"
+
+type Provider struct {
+	secretsFilePath string
+}
+
+func NewProvider(secretsFilePath string) (model.Provider, error) {
```

Comment:
I'm still not sure why this is a constructor argument. It should either be a base path or rather an `fs.Fs` interface.

Here is how I expect this to work:

`MY_ENV_VAR=file://path/to/file`

`MY_ENV_VAR` should be replaced with the contents of the file.

This isn't the only way the file provider could work (eg. we could support and parse YAML files), but I would expect the file path to come from the input.

If that's not the expected behavior for some reason, please explain why (eg. in doc comment).

Reply:
Note here @sagikazarmark, we discussed a different approach in the review above https://github.com/bank-vaults/secret-init/pull/17#discussion_r1392454226 because it did not make much sense the way it was working before

Reply:
Please explain it in a doc comment/documentation somewhere.

Reply:
I reworked the logic, according to your feedback, documented it in the PR's description: #17